swatch-product-configuration/src/main/java/com/redhat/swatch/configuration/registry/Variant.java

```diff
@@ -32,11 +34,25 @@
  * Variants all have the same billing model.
  */
 @Data
-@Builder
 public class Variant {
 
   @NotNull @NotEmpty private String tag; // required
-  private List<String> roles;
-  private List<String> engineeringIds;
-  private List<String> productNames;
+  private List<String> roles = new ArrayList<>();
+  private List<String> engineeringIds = new ArrayList<>();
+  private List<String> productNames = new ArrayList<>();
+
+  public Optional<Subscription> getSubscription() {
+    return SubscriptionConfigRegistry.getInstance().lookupSubscriptionByTag(tag);
+  }
+
```

Comment:
The important part of this "lookup methods" cards was less about getting every necessary lookup method implemented, and more about getting the values read from disk and putting the pattern of lookup methods in place.

Had a discussion with @kahowell last week and decided that I was going to put the PR up for the ones I already defined.  As the other cards in the epic get worked on, lookup methods can be added as needed.  Coincidently, I'm  implementing the findByEngProductId in the next card because I ended up needing it.

I'll remove the two methods you mentioned from the JIRA card description.